sources/VT100Screen.m

```diff
@@ -848,13 +848,22 @@ - (void)clearBuffer {
 // This clears the screen, leaving the cursor's line at the top and preserves the cursor's x
 // coordinate. Scroll regions and the saved cursor position are reset.
 - (void)clearAndResetScreenPreservingCursorLine {
+    int numberOfLinesInPrompt = [iTermAdvancedSettingsModel numberOfLinesInPrompt];
```

Comment:
This is very good remark. I didn't use shell integration, and I think a lot about how to automate this (to do not use `numberOfLinesInPrompt` advanced setting).

I will add shell integration support, thank you!